### DISCUSSION TITLE
Can zeek be used for TCP network stack's conformance test?

### DISCUSSION THREAD

**Discussion by kittu-57** [zeek/zeek#3419] [How-Tos]
Hi Team,

I am in quest of finding the right set of tools that confirms Linux OS's TCP network stack in my project.
Idea is to find out if there are any anomalies present in the protocol code.

I happened to give a try to use tcptest-by-google([http://code.google.com/p/tcptest/downloads/list]) which seems to be outdated with no support to use it in Linux.

Any pointers to get the appropriate tool to test all basic compliance tests of TCP, is very much appreciated.

Thank you for your time.

--
Krishna K

**Comment by vpax** [comment]
Unfortunately Zeek is not an apt tool for finding network stack issues. In particular, Zeek has to deal (to varying degrees) with a number of artifacts that arise from problems with packet capture, and it can be difficult to distinguish between those and actual stack problems. Zeek is fairly robust to a number of problems, and thus its analysis will mask their presence rather than reveal it.
